miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -250,7 +244,8 @@ public void applySetChangesContacts(List<ProjectContact> to, List<ProjectContact
         toItem -> from.stream().noneMatch(fromItem -> fromItem.getContact().getId() == toItem.getContact().getId()
             && fromItem.getContactRole().getId() == toItem.getContactRole().getId()));
     from.forEach(fromItem -> {
-      if (to.stream().noneMatch(toItem -> toItem.getContact().getId() == fromItem.getContact().getId())) {
+      if (to.stream().noneMatch(toItem -> toItem.getContact().getId() == fromItem.getContact().getId()
+          && fromItem.getContactRole().getId() == toItem.getContactRole().getId())) {
```

Comment:
Can move the contactId+contactRoleId checking to a `private static boolean contactsMatch(ProjectContact a, ProjectContact b)` method to use here and the other 2 places you're checking the same (removeIf just above and in validation)